bin/publish-modules

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+
+var exec = require( 'child-process-promise' ).exec;
+var semver = require( 'semver' );
+
+console.log( 'Modules requiring publish:' );
+
+function getAllModules() {
+	var command = 'find client -type f -print | grep \'package.json\'';
+	return exec( command ).then( function( result ) {
+		return result.stdout.split( /\s+/ );
```

Comment:
Partially resolved in: https://github.com/Automattic/wp-calypso/commit/e9f71b541167882fdca1822c6f2de5fff4f737aa

The error messaging will reveal if there are any paths that do not resolve properly. If a path has spaces or improper characters then it will error out and the issue can be resolved.